ptf/warm_boot/warm_saiswitch.py

```diff
@@ -0,0 +1,73 @@
+# Copyright (c) 2021 Microsoft Open Technologies, Inc.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+#
+#    THIS CODE IS PROVIDED ON AN *AS IS* BASIS, WITHOUT WARRANTIES OR
+#    CONDITIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT
+#    LIMITATION ANY IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS
+#    FOR A PARTICULAR PURPOSE, MERCHANTABILITY OR NON-INFRINGEMENT.
+#
+#    See the Apache Version 2.0 License for specific language governing
+#    permissions and limitations under the License.
+#
+#    Microsoft would like to thank the following companies for their review and
+#    assistance with these files: Intel Corporation, Mellanox Technologies Ltd,
+#    Dell Products, L.P., Facebook, Inc., Marvell International Ltd.
+
+from warm_test_utils import *  # pylint: disable=wildcard-import; lgtm[py/polluting-import]
+
+"""
+Thrift SAI interface Switch tests
+"""
+
+class WarmAvailableIPv4RouteEntryTest(AvailableIPv4RouteEntryTest):
+    """
+    Warm boot Test for available IPv4 Route Entry.
+    """
+    @warm_setup
+    def setUp(self):
+        print("setUp WarmAvailableIPv4RouteEntryTest")
+        AvailableIPv4RouteEntryTest.setUp(self)
+
+
+    def setUp_starting(self):
+        print("Skip setUp_starting WarmAvailableIPv4RouteEntryTest")
+
+
+    def setUp_post_start(self):
+        print("setUp_post_start WarmAvailableIPv4RouteEntryTest")
+        SaiHelperBase.setUp(self)
```

Comment:
why is this setUp call required here?

Reply:
We have three mode, pre-start, starting, post-start, 
in starting, we will not use any SAI_PTF apis to handle the swtich, as the sai sdk doesn't start.
in post start, we expect we can call some sai apis.

Then, after the reboot, we also need to mapping the data with the test framework, including remap the bridge port id.
then take bridge for example;
- for  the devices that need to recreated we will make more operation in this step, as those data should be consistent with the restored data from warmboot, (will make it when start landing those kinds of device for testing)
- for other platform don't need to recreate the bridge, we can use those setup to remap them again.

then in the port_start test, we can use those data to handle and check switch status with SAI-PTF apis

Reply:
Got it. Thanks for the explanation. Is it possible to do this automatically in the base class instead of having every warm boot class do this?

Reply:
Yes, this could be and it is a good idea.

But, I have some concerns about the refactor from the fundamental layer, so choose to wrap some methods in the test class and introduce few modifications to the original test class. Then, just re-orgainze the method in a new place.

Apart from that, this is also a proposal for reusing the existing test cases, I will try to make some experiments with this new structure and see if there any other improvements could be.

thanks.